Batches 4.19 Cloud Compute bugs (#94607)
openshift/openshift-docs: release_notes/ocp-4-19-release-notes.adoc

[id="ocp-release-note-cloud-compute-bug-fixes_{context}"]
==== Cloud Compute

> * When upgrading {gcp-short} clusters that use a boot disk that is not compatible with UEFI, you cannot enable shielded VM support. 

Reviewer: ```suggestion
* When upgrading {gcp-short} clusters that use a boot disk that is not compatible with UEFI, you cannot enable shielded-VM support. 
```
Does everyone know what a Unified Extensible Firmware Interface is?

Author: In this context, I think yeah

> Previously, this prevented the creation of new machines. 

Reviewer: ```suggestion
Previously, this prevented the creation of new machines. 
```
virtual machines? compute machines?

> With this release, disks with known UEFI incompatiblity have shielded VM support disabled. 

Reviewer: ```suggestion
With this release, disks with known UEFI incompatiblity have shielded-VM support disabled. 
```
I assume we mean "support for a shielded VM", as opposed to VM support that is shielded.

Author: It's actualyl a feature name - I'll check but I think it might be "Shielded VM". Good catch!

This primarily affects customers upgrading from {product-title} version 4.12 to 4.13 using the {gcp-short} marketplace images. 
(link:https://issues.redhat.com/browse/OCPBUGS-17079[OCPBUGS-17079])

> * Previously, virtual machines (VMs) in a cluster that ran on {azure-short} failed because the attached network interface controller (NIC) was in a `ProvisioningFailed` state. 

Reviewer: ```suggestion
* Previously, VMs in a cluster that ran on {azure-short} failed because the attached network interface controller (NIC) was in a `ProvisioningFailed` state. 
```
We didn't define it in the preceding note; VM is _probably_ defined way above in the release notes by now. Either I'd define it in the first bullet of this section, or not at all as assume already used.

With this release, the Machine API controller checks the provisioning status of a NIC and refreshes the VMs on a regular basis to prevent this issue. 
(link:https://issues.redhat.com/browse/OCPBUGS-31515[OCPBUGS-31515])

> * Previously, the certificate signing request (CSR) approver included certificates from other systems within its calculations for whether it was overwhelmed and should stop approving certificates. 

Reviewer: IDK why these longer bugs start so many newlines? It looks fine in the preview, but you might want to check your editor just in case. (Might just be this PR preview for me, who knows?)

Author: I have taken to breaking lines at the end of a sentence for easier diffs :)

> * Previously, the certificate signing request (CSR) approver included certificates from other systems within its calculations for whether it was overwhelmed and should stop approving certificates. 

Reviewer: ```suggestion
* Previously, in larger clusters that had other subsystems using CSRs, the certificate signing request (CSR) approver counted unrelated, unapproved CSRs towards its total and prevented further approvals.
```

> In larger clusters, with other subsystems using CSRs, the CSR approver counted unrelated unapproved CSRs towards its total and prevented further approvals. 

Reviewer: ```suggestion
```

With this release, the CSR approver uses a `signerName` property as a filter and only includes CSRs that it can approve. 
As a result, the CSR approver only prevents new approvals when there are a large number of unapproved CSRs for the relevant `signerName` values. 
(link:https://issues.redhat.com/browse/OCPBUGS-36404[OCPBUGS-36404])

> * Previously, the Machine API controller read the zone number to populate machine zone information. 

Reviewer: ```suggestion
* Previously, the Machine API controller read only the zone number to populate machine zone information. 
```

For machines in {azure-short} regions that only support availability sets, the set number represents the zone, so the Machine API controller did not populate their zone information. 
With this release, the Machine API controller references the {azure-short} fault domain property.
This property works for availability sets and availability zones, so the controller correctly reads the fault domain in each case and machines always report a zone. 
(link:https://issues.redhat.com/browse/OCPBUGS-38570[OCPBUGS-38570])

* Previously, increased granularity in {gcp-short} zone API error messages caused the machine controller to mistakenly mark some machines with invalid configurations as valid with a temporary cloud error. 

> This behavior prevented the invalid machine from transitioning to a failed state. 

Reviewer: ```suggestion
This behavior prevented the invalid machines from transitioning to a failed state. 
```
existential machine crisis here

With this release, the machine controller handles the more granular error messages correctly so that machines with an invalid zone or project ID correctly move to a failed state. 
(link:https://issues.redhat.com/browse/OCPBUGS-43531[OCPBUGS-43531])

* Previously, some permissions required for linked actions were missing. 
Linked actions create the subresources necessary for other {azure-short} resources that the cloud controller manager and {product-title} require. 

> With this release, the cloud controller manager for {azure-short} has the following additional permissions:

Reviewer: ```suggestion
With this release, the cloud controller manager for {azure-short} has the following permissions for linked actions:
```
I think? Got a bit confused by the verbiage here.

+
--
** `Microsoft.Network/applicationGateways/backendAddressPools/join/action`
** `Microsoft.Network/applicationSecurityGroups/joinIpConfiguration/action`
** `Microsoft.Network/applicationSecurityGroups/joinNetworkSecurityRule/action`
** `Microsoft.Network/ddosProtectionPlans/join/action`
** `Microsoft.Network/gatewayLoadBalancerAliases/join/action`
** `Microsoft.Network/loadBalancers/backendAddressPools/join/action`
** `Microsoft.Network/loadBalancers/frontendIPConfigurations/join/action`
** `Microsoft.Network/loadBalancers/inboundNatRules/join/action`
** `Microsoft.Network/networkInterfaces/join/action`
** `Microsoft.Network/networkSecurityGroups/join/action`
** `Microsoft.Network/publicIPAddresses/join/action`
** `Microsoft.Network/publicIPPrefixes/join/action`
** `Microsoft.Network/virtualNetworks/subnets/join/action`
--
+
(link:https://issues.redhat.com/browse/OCPBUGS-44126[OCPBUGS-44126])

* Previously, some permissions required for linked actions were missing. 
Linked actions create the subresources necessary for other {azure-short} resources that the Machine API and {product-title} require. 

> With this release, the Machine API provider for {azure-short} has the following additional permissions:

Reviewer: ```suggestion
With this release, the Machine API provider for {azure-short} has the following additional permissions for linked actions:
```

+
--
** `Microsoft.Compute/disks/beginGetAccess/action`
** `Microsoft.KeyVault/vaults/deploy/action`
** `Microsoft.ManagedIdentity/userAssignedIdentities/assign/action`
** `Microsoft.Network/applicationGateways/backendAddressPools/join/action`
** `Microsoft.Network/applicationSecurityGroups/joinIpConfiguration/action`
** `Microsoft.Network/applicationSecurityGroups/joinNetworkSecurityRule/action`
** `Microsoft.Network/ddosProtectionPlans/join/action`
** `Microsoft.Network/gatewayLoadBalancerAliases/join/action`
** `Microsoft.Network/loadBalancers/backendAddressPools/join/action`
** `Microsoft.Network/loadBalancers/frontendIPConfigurations/join/action`
** `Microsoft.Network/loadBalancers/inboundNatPools/join/action`
** `Microsoft.Network/loadBalancers/inboundNatRules/join/action`
** `Microsoft.Network/networkInterfaces/join/action`
** `Microsoft.Network/networkSecurityGroups/join/action`
** `Microsoft.Network/publicIPAddresses/join/action`
** `Microsoft.Network/publicIPPrefixes/join/action`
** `Microsoft.Network/virtualNetworks/subnets/join/action`
--
+
(link:https://issues.redhat.com/browse/OCPBUGS-44130[OCPBUGS-44130])

> * Previously, installing an {aws-short} cluster failed in certain environments on existing subnets when the `publicIp` parameter in the compute machine set custom resource was set to `false`. 

Reviewer: ```suggestion
* Previously, installing an {aws-short} cluster failed in certain environments on existing subnets when the `publicIp` parameter in the compute machine set CR was set to `false`. 
```
probably defined way previously in RNs?

> With this release, a fix ensures that a configuration value set for `publicIp` no longer causes issues when the installation program provisions machines for your {aws-short} cluster in certain environment.  

Reviewer: ```suggestion
With this release, a fix ensures that a configuration value set for `publicIp` no longer causes issues when the installation program provisions machines for your {aws-short} cluster in certain environments.  
```

(link:https://issues.redhat.com/browse/OCPBUGS-44373[OCPBUGS-44373])

* Previously, {gcp-short} clusters that used non-UEFI disks failed to load. 
This release adds a check to ensure that disks are UEFI-compatible before enabling features that require UEFI, such as secure boot. 

> As a result, you can use non-UEFI disks if needed. 

Reviewer: but they then have a limited feature set? this conclusion kind of threw me here

> This change adds new permissions requirements for `compute.images.get` and `compute.images.getFromFamily`. (link:https://issues.redhat.com/browse/OCPBUGS-44671[OCPBUGS-44671])

Reviewer: ```suggestion
This change adds new permission requirements for `compute.images.get` and `compute.images.getFromFamily`. (link:https://issues.redhat.com/browse/OCPBUGS-44671[OCPBUGS-44671])
```
not tracking how this follows the not-that-device-maybe-that-device logic of the bug

* Previously, when the {aws-short} DHCP option set was configured to use a custom domain name that contains a trailing period (`.`), {product-title} installation failed. 
With this release, the logic that extracts the hostname of EC2 instances and turns them into kubelet node names trims trailing periods so that the resulting Kubernetes object name is valid. 

> Trailing periods in the DHCP option set no longer cause installation to fail. (link:https://issues.redhat.com/browse/OCPBUGS-45306[OCPBUGS-45306])

Reviewer: ```suggestion
Trailing periods in the DHCP network protocol set no longer cause installation to fail. (link:https://issues.redhat.com/browse/OCPBUGS-45306[OCPBUGS-45306])
```
just have never seen DHCP referred to as an "option set" before; seemed odd and non-specific

* Previously, the number of {azure-short} availability set fault domains used a fixed value of `2`. 
This setting works in most {azure-short} regions because fault domain counts are typically at least 2. 

> However, this setting failed in the `centraluseuap` and `eastusstg` regions. 

Reviewer: implying that these areas had lower values that caused some kind of failure?

With this release, the number of availability set fault domains in a region is set dynamically. 
(link:https://issues.redhat.com/browse/OCPBUGS-45663[OCPBUGS-45663])

> * Previously, the {azure-short} cloud controller manager panicked when there was a temporary API server disconnection. 

Reviewer: what's the user reality of a panicked CCM?

With this release, the {azure-short} cloud controller manager correctly recovers from temporary disconnection. 
(link:https://issues.redhat.com/browse/OCPBUGS-45859[OCPBUGS-45859])

* Previously, some services became stuck in a pending state due to incorrect or missing annotations. 
With this release, validation added to the {azure-short} `service.beta.kubernetes.io/azure-load-balancer-tcp-idle-timeout` and {gcp-short} `cloud.google.com/network-tier` annotations resolves the issue. 
(link:https://issues.redhat.com/browse/OCPBUGS-48481[OCPBUGS-48481])

* Previously, the method used to fetch the provider ID from {aws-short} could fail to provide this value to the kubelet when needed. 
As a result, sometimes machines could get stuck in different states and fail to complete initialization. 

> With this release, the provider ID is consistently set on kubelet startup. 

Reviewer: ```suggestion
With this release, the provider ID is consistently set when the kubelet starts up. 
```

(link:https://issues.redhat.com/browse/OCPBUGS-50905[OCPBUGS-50905])

* Previously, an incorrect endpoint in the {azure-short} cloud controller manager caused installations on {azure-full} Government Cloud to fail. 
The issue is resolved in this release.
(link:https://issues.redhat.com/browse/OCPBUGS-50969[OCPBUGS-50969])

* Previously, the Machine API sometimes detected an unhealthy control plane node during cluster creation on {ibm-cloud-title} and attempted to replace the node. 
This effectively destroyed the cluster. 

> With this release, the Machine API only attempts to replace unhealthy compute nodes during cluster creation. 

Reviewer: ```suggestion
With this release, the Machine API only attempts to replace unhealthy compute nodes during cluster creation and does not attempt to replace unhealthy control plane nodes. 
```
just because I had to read that twice

(link:https://issues.redhat.com/browse/OCPBUGS-51864[OCPBUGS-51864])

* Previously, {azure-short} spot machines that were evicted before their node became ready could get stuck in the `provisioned` state. 

> With this release, {azure-short} spot instances now use a delete eviction policy. 

Reviewer: ```suggestion
With this release, {azure-short} spot instances now use a delete-eviction policy. 
```

[id="ocp-release-note-image-registry-bug-fixes_{context}"]
==== Registry

* Previously, image importing from blocked registries would fail if those registries were configured with `NeverContactSource`, even when mirror registries were set up. With this update, image importing is no longer blocked when a registry has mirrors configured. This ensures that image imports succeed even if the original source was set to `NeverContactSource` in the `ImageDigestMirrorSet` or `ImageTagMirrorSet` resources. (link:https://issues.redhat.com/browse/OCPBUGS-44432[OCPBUGS-44432])

[discrete]
[id="ocp-release-note-installer-bug-fixes_{context}"]
==== Installer

> * Previously, if you attempted to install an {aws-first} cluster with minimum privileges and you did not specify an instance type in the `install-config.yaml` file, installation of the cluster failed. This issue happened because the installation program could not find supported instance types that the cluster uses in availability zones. For example, the `m6i.xlarge` default instance type was unavailable in `ap-southeast-4` and `eu-south-2` availability zones. With this release, the `openshift-install` program now requires the `ec2:DescribeInstanceTypeOfferings` {aws-short} permission to prevent the installation of the cluster from failing in situations where `m6i.xlarge` or another supported instance type is unavailable in a supported availability zone. (link:https://issues.redhat.com/browse/OCPBUGS-46596[OCPBUGS-46596])

Reviewer: ```suggestion
* Previously, if you attempted to install an {aws-first} cluster with minimum privileges and you did not specify an instance type in the `install-config.yaml` file, installation of the cluster failed. This issue happened because the installation program could not find supported instance types that the cluster could use in supported availability zones. For example, the `m6i.xlarge` default instance type was unavailable in `ap-southeast-4` and `eu-south-2` availability zones. With this release, the `openshift-install` program now requires the `ec2:DescribeInstanceTypeOfferings` {aws-short} permission to prevent the installation of the cluster from failing in situations where `m6i.xlarge` or another supported instance type is unavailable in a supported availability zone. (link:https://issues.redhat.com/browse/OCPBUGS-46596[OCPBUGS-46596])
```
